### DISCUSSION TITLE
Test require distinction between floats and integers

### DISCUSSION THREAD

**Discussion by Patitotective** [kdl-org/kdl#300] [General]
Quoting the spec:
> _There is no logical distinction in KDL between real numbers, integers, and floating point numbers. It's up to individual implementations to determine how to represent KDL numbers._

The spec says that there is no distinction between integer and float numbers, though some tests like `numeric_prop.kdl` and `positive_int.kdl` need your KDL implementation to distinguish between both floats and integers in order to match the expected_kdl.

---
`numeric_prop.kdl`:
- Input: `node prop=10.0`
- Expected `node prop=10.0`

`positive_int.kdl`:
- Input: `node +10`
- Expected `node 10`

(As you can see an integer as input expects an integer as output, same with float, this makes your implementation to difference integers and floats).

I suggest making all tests expect floating numbers.

**Comment by zkat** [comment]
it's expected that implementations will need to massage their test executions to match the tests. The tests are meant to be general guidelines, and if your parser has particular quirks, you can definitely compensate for that in your own test suite. I think this is a good example of that.
